### ISSUE TITLE
same old, same old:  pip install fails

### ISSUE THREAD

**Issue by lewisl** [PyTables/PyTables#385]
Must be the most searched topic about hdf5.  

I have installed:  numpy 1.8.2, numexpr 2.4, hdf5 1.8.13.
I have exported the environment variable HDF5_DIR=/usr/local.  And, yes it echoes.
Still, pip running the build can't find the hdf5 libraries.

You should this option:  pip install --install-option='--prefix=<PREFIX>' \
-e git+https://github.com/PyTables/PyTables.git@v.2.3#egg=tables

Uh, but what should prefix be?  How about an example or at least text that says what prefix should refer to?  Perhaps the directory for hdf5?   

This is the MOST widely reported problem with tables and the problem continues.

**Comment by eraoul** [comment]
I'm with lewisl on this one.

I'm not using "homewbrew" since it causes all sorts of other troubles for me.

Just trying to do a clean install of hdf5 and tables.

I build hdf5 from source.

I was hoping that "pip install" would work, but as far as I can tell it's fubar, at least on mac.

This does not work: sudo pip install --install-option="--hdf5=/usr/local/hdf5" tables

Any reason the hdf5 option is ignored here, as is the HDF5_DIR environment var??

Perhaps the only way to install on mac is from source; if so I think the docs should be updated. Assuming pip never works for tables on mac unless someone says otherwise.

**Comment by andreabedini** [comment]
Hi @eraoul, using homebrew is definitely not a requirement. Everything is supposed to work if you build HDF5 yourself.

> Perhaps the only way to install on mac is from source

Since we don't publish binary wheels but only the source package, `pip install` will install from source. There's a caveat: pip will cache wheels built from source packages. Check the pip documentation if you need to disable the caching.

> Any reason the hdf5 option is ignored here, as is the HDF5_DIR environment var??

If this is the case, it's a bug.

@eraoul Could you please attach a log of the messages you get? I assume HDF5 is correctly installed in `/usr/local/hdf5`.

**Comment by jpcofr** [comment]
Hi... I'm facing the same problem as @eraoul on OS X. I ran 
`pip install --install-option="--hdf5=/usr/local/Cellar/hdf5" tables` and got 

> Collecting tables
>   Using cached tables-3.2.3.1.tar.gz
> Requirement already satisfied (use --upgrade to upgrade): numpy>=1.8.0 in /usr/local/lib/python2.7/site-packages (from tables)
> Requirement already satisfied (use --upgrade to upgrade): numexpr>=2.5.2 in /usr/local/lib/python2.7/site-packages (from tables)
> Skipping bdist_wheel for tables, due to binaries being disabled for it.
> Installing collected packages: tables
>   Running setup.py install for tables: started
>     Running setup.py install for tables: finished with status 'error'
>     Complete output from command /usr/local/opt/python/bin/python2.7 -u -c "import setuptools, tokenize;__file__='/private/var/folders/q1/dt_phwt515l7v8mpck3stxtr0000gp/T/pip-build-9idOY8/tables/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /var/folders/q1/dt_phwt515l7v8mpck3stxtr0000gp/T/pip-P5u5vv-record/install-record.txt --single-version-externally-managed --compile --hdf5=/usr/local/Cellar/hdf5:
>     * Using Python 2.7.12 (default, Jun 29 2016, 14:05:02)
>     * USE_PKGCONFIG: True
>     /var/folders/q1/dt_phwt515l7v8mpck3stxtr0000gp/T/H5close9Ym8lX.c:1:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
>     main (int argc, char **argv) {
>     ^
>     /var/folders/q1/dt_phwt515l7v8mpck3stxtr0000gp/T/H5close9Ym8lX.c:2:5: warning: implicit declaration of function 'H5close' is invalid in C99 [-Wimplicit-function-declaration]
>         H5close();
>         ^
>     2 warnings generated.
>     .. ERROR:: Could not find a local HDF5 installation.
>        You may need to explicitly state where your local HDF5 headers and
>        library can be found by setting the ``HDF5_DIR`` environment
>        variable or by using the ``--hdf5`` command-line option.
>     
>     ----------------------------------------
>

**Comment by jpcofr** [comment]
I installed hdf5 using homebrew and I gave the complete path (including the hdf5 version). Afterwards, I ran
` env HDF5_DIR='/usr/local/Cellar/hdf5/1.8.16_1' pip install tables`

And I successfully installed tables-3.2.3.1

**Comment by EricDeveaud** [comment]
Hello, 

I have a problem with tables installation with non standard hdf5 installation directory
hdf5 is installed in /local/gensoft2/lib/hdf5/1.8.14 and is loaded via a modulefile that set and env var HDF5_ROOT 

python-3.6.0 is also available thru a modulefile

I have performed as pointed by @jpcofr using ```HDF5_DIR="$HDF5_ROOT" pip3 install --user tables```

see:
```
tars-submit0:~ > env HDF5_DIR="$HDF5_ROOT" pip3 install --user tables
Collecting tables
  Using cached tables-3.3.0.tar.gz
Requirement already satisfied: numpy>=1.8.0 in /mount/gensoft2/exe/Python/3.6.0/lib/python3.6/site-packages (from tables)
Requirement already satisfied: numexpr>=2.5.2 in ./.local/lib/python3.6/site-packages (from tables)
Requirement already satisfied: six in ./.local/lib/python3.6/site-packages (from tables)
Installing collected packages: tables
  Running setup.py install for tables ... done
Successfully installed tables-3.3.0
```

pip installation ended succesfully 

but when I try to import tables in python3 I have the following error:

```
tars-submit0:~ > python3
Python 3.6.0 (default, Feb 21 2017, 10:42:53) 
[GCC 4.9.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tables
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/pasteur/homes/edeveaud/.local/lib/python3.6/site-packages/tables/__init__.py", line 90, in <module>
    from .utilsextension import (
ImportError: libhdf5.so.9: cannot open shared object file: No such file or directory
>>> 
```

what surprised me is that all libraries in .local/lib/python3.6/site-packages/tables are not linked to libhdf5

see:
```
tars-submit0:~ > ldd .local/lib/python3.6/site-packages/tables/*.so | grep hdf5
	libhdf5.so.9 => not found
.local/lib/python3.6/site-packages/tables/hdf5extension.cpython-36m-x86_64-linux-gnu.so:
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
```

any ideas ?

**Comment by tomkooij** [comment]
There are wheels available for both Mac OSX, Linux and Windows, that come with a combined hdf5 library. These wheels work out of the box using `pip`.

Installation with pip from the sdist (`tables-3.3.0.tar.gz`) might be easier by extracting the tar and running `python setup.py install` from the folder, which should give much more info about the build. That should detect and link against the installed HDF5 installation. If not, please copy a full log of `python setup.py install` here.

**Comment by avalentino** [comment]
> pip installation ended succesfully
>
> but when I try to import tables in python3 I have the following error:

@EricDeveaud if the installation step works correctly then you probably just need to set the LD_LIBRARY_PATH to include also  the path where libhdf5.so.9 is located, i.e.

 `/local/gensoft2/lib/hdf5/1.8.14`

or maybe

`/local/gensoft2/lib/hdf5/1.8.14`

**Comment by EricDeveaud** [comment]
@tomkooij 
using pip3 (python 3.6) and issuing pip install tables, as described in the documentation , lead to a non functional installation. 
see the ldd trace above. 
```
wget https://pypi.python.org/packages/97/eb/ea2102f5a210a62f9f7387cf9912cb841f4a9089dbb232e642daa2626769/tables-3.3.0.tar.gz
tar xf tables-3.3.0.tar.gz
cd tables-3.3.0
module load Python/3.6.0 hdf5/1.8.14 
export PREFIX=~/.local
python3 setup.py install --prefix=$PREFIX --lflags="-Xlinker -rpath -Xlinker $HDF5_ROOT/lib | tee log"
```
log is attached 

and I still have:
```
tars-submit0:~/tables-3.3.0 > ldd ~/.local/lib/python3.6/site-packages/tables-3.3.0-py3.6-linux-x86_64.egg/tables/*.so | grep hdf5
	libhdf5.so.9 => not found
/pasteur/homes/edeveaud/.local/lib/python3.6/site-packages/tables-3.3.0-py3.6-linux-x86_64.egg/tables/hdf5extension.cpython-36m-x86_64-linux-gnu.so:
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
	libhdf5.so.9 => not found
```
that lead to the same error

I was abble to have a  functional installation using the following:
```
python3 setup.py install --lflags="-Wl,-rpath,$HDF5_ROOT/lib" --prefix=$PREFIX | tee log2
```
no more libhdf5.so link problem and import tables under python is OK 

unfortunatly I am not abble to pass the correct ```--lflags``` to pip. 

@avalentino 
LD_LIBRARY_PATH is not the solution. I tend to avoid LD_LIBRARY_FLAGS as much as possible, and use it only at the last solution. 
in a wrapper just before execution of the software, I may accept, but exporting it to the environment is a no way for me
I won't be exporting  LD_LIBRARY_FLAGS in our user env. we have >600 soft//packages installed.

**Comment by tomkooij** [comment]
@EricDeveaud: Looking at your description it seems to be a `setup.py` problem: I'll look into it. Thanks for the log.

**Comment by tomkooij** [comment]
@EricDeveaud 

Your log indicates the shared lib `libhdf5.so.x` cannot be found by setuptools:
```
.. WARNING:: Could not find the HDF5 runtime.
   The HDF5 shared library was *not* found in the default library
   paths. In case of runtime problems, please remember to install it.
```
Is `libhdf5.so.x` in PATH?

If that fails you can pass `--flags` through `pip` to `setup.py` using  the `LFLAGS` enviroment var.

**Comment by EricDeveaud** [comment]
yep libhdf5 is available but not in std location,  and is found by the setup.py.

``` Found HDF5 headers at ``/local/gensoft2/lib/hdf5/1.8.14/include``, library at ``/local/gensoft2/lib/hdf5/1.8.14/lib``.
```

module hdf5 set the following 
```
module show hdf5
-------------------------------------------------------------------
/local/gensoft2/devmodules/hdf5/1.8.14:

module-whatis    Set environnement for hdf5 (1.8.14) 
module-whatis    topic_3316 
prepend-path     LIBRARY_PATH /local/gensoft2/lib/hdf5/1.8.14/lib 
prepend-path     LD_RUN_PATH /local/gensoft2/lib/hdf5/1.8.14/lib 
prepend-path     -d   LDFLAGS -L/local/gensoft2/lib/hdf5/1.8.14/lib 
prepend-path     CMAKE_LIBRARY_PATH /local/gensoft2/lib/hdf5/1.8.14/lib 
prepend-path     C_INCLUDE_PATH /local/gensoft2/lib/hdf5/1.8.14/include 
prepend-path     CPLUS_INCLUDE_PATH /local/gensoft2/lib/hdf5/1.8.14/include 
prepend-path     CMAKE_INCLUDE_PATH /local/gensoft2/lib/hdf5/1.8.14/include 
prepend-path     -d   CPPFLAGS -I/local/gensoft2/lib/hdf5/1.8.14/include 
prepend-path     PATH /local/gensoft2/lib/hdf5/1.8.14/bin 
setenv           HDF5_ROOT /local/gensoft2/lib/hdf5/1.8.14 

```

**Comment by tomkooij** [comment]
The warning `Could not find HDF5 runtime` means it cannot find the runtime location of `libhdf5.so` (i.e. it is found in.
 
I think `/local/gensoft2/lib/hdf5/1.8.14/lib` should be in your PATH.

To test start python:
```
>>> from ctypes import CDLL
>>>CDLL('libhdf5.so')
```
This should return an object without an `OSError` exception. If you get the `OSError` your libary cannot be loaded by CPython unrelated to PyTables.

**Comment by EricDeveaud** [comment]
maybee I'm missing something.

```
Python 3.6.0 (default, Feb 21 2017, 10:42:53) 
[GCC 4.9.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from ctypes import CDL
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'CDL'
>>> 
```

**Comment by tomkooij** [comment]
`from ctypes import CDLL`

**Comment by andreabedini** [comment]
Warning: all this is before my  ☕️ 

First: @EricDeveaud your modulefile should set the `LD_LIBRARY_PATH` for you. This is what I have on my Cray (CRAY_LD_LIBRARY_PATH)

```
andreabedini@magnus-2:~> module show hdf5
-------------------------------------------------------------------
/opt/cray/modulefiles/hdf5/1.8.11:

conflict	 hdf5-parallel 
conflict	 cray-hdf5-parallel 
conflict	 cray-hdf5 
conflict	 PrgEnv-pathscale 
prepend-path	 PE_PKGCONFIG_PRODUCTS PE_HDF5 
setenv		 PE_HDF5_VOLATILE_PRGENV GNU 
setenv		 PE_HDF5_VOLATILE_PKGCONFIG_PATH /opt/cray/hdf5/1.8.11/@PRGENV@/@PE_HDF5_GENCOMPS@/lib/pkgconfig 
setenv		 PE_HDF5_GENCOMPS_GNU 48 47 
setenv		 PE_HDF5_FIXED_PRGENV CRAY INTEL 
prepend-path	 PE_CRAY_FIXED_PKGCONFIG_PATH /opt/cray/hdf5/1.8.11/CRAY/81/lib/pkgconfig 
prepend-path	 PE_INTEL_FIXED_PKGCONFIG_PATH /opt/cray/hdf5/1.8.11/INTEL/130/lib/pkgconfig 
prepend-path	 PE_PKGCONFIG_LIBS hdf5:hdf5_hl 
prepend-path	 PE_FORTRAN_PKGCONFIG_LIBS hdf5hl_fortran:hdf5_fortran 
prepend-path	 PE_CXX_PKGCONFIG_LIBS hdf5_hl_cpp:hdf5_cpp 
setenv		 CRAY_HDF5_VERSION 1.8.11 
setenv		 CRAY_HDF5_DIR /opt/cray/hdf5/1.8.11 
setenv		 PE_HDF5_DIR /opt/cray/hdf5/1.8.11 
prepend-path	 PATH /opt/cray/hdf5/1.8.11/bin 
setenv		 HDF5_DIR /opt/cray/hdf5/1.8.11/CRAY/81 
setenv		 HDF5_INCLUDE_OPTS /opt/cray/hdf5/1.8.11/CRAY/81/include 
prepend-path	 CRAY_LD_LIBRARY_PATH /opt/cray/hdf5/1.8.11/CRAY/81/lib
-------------------------------------------------------------------
```

I see that your modulefile sets `LD_RUN_PATH`, which AFAIK is equivalent to passing `-rpath` to the (GNU at least) linker, but I am not sure.

Second: @EricDeveaud is [log.txt](https://github.com/PyTables/PyTables/files/795995/log.txt) created with `--lflags="-Wl,-rpath,$HDF5_ROOT/lib"` because the linker gets this `-Xlinker -rpath -Xlinker HDF5_ROOT/lib` flag that confuses me. Maybe somehow `$HDF5_ROOT` doesn't get expanded?

Third: I agree the messages are confusing and `setup.py` should be rewritten from scratch but this is not a bug, it's the way unix work. Passing `HDF5_DIR` to `setup.py` makes it pass the flag `-L/local/gensoft2/lib/hdf5/1.8.14/lib` to the linker, which successfully finds the library and compilation succeed. When you do `import tables`, the *runtime linker* needs to find the shared library and it looks in the paths in $LD_LIBRARY_PATH (plus a few standard ones I believe). Without that it won't find the library and the program won't run. @EricDeveaud kudos for thinking about using `-rpath` which I believe is your only alternative to `LD_LIBRARY_PATH`. The way to give `pip` options to pass to `setup.py` is [in the docs](https://pip.pypa.io/en/stable/reference/pip_install/#cmdoption--install-option).

tl;dr (I haven't double checked the quotes escaping):

```
pip --install-options="--lflags=\"-Wl,-rpath,$HDF5_ROOT/lib\" --prefix=$PREFIX" tables
```

**Comment by avalentino** [comment]
@andreabedini +1

> pip --install-options="--lflags=\"-Wl,-rpath,$HDF5_ROOT/lib\" --prefix=$PREFIX" tables 

Probably adding the above example to the documentation is a good idea

**Comment by codeisnotcode** [comment]
I tried all these suggestions but still was unable to get an installation, so I just re-wrote my code to eliminate the use of tables. I agree with the above comments that this should be as simple as pip install tables, and at least in my case, it resulted in me avoiding the package altogether.

**Comment by andreabedini** [comment]
> I agree with the above comments that this should be as simple as pip install tables

We all agree here (developers included)

**Comment by FrancescAlted** [comment]
@codeisnotcode Thanks for reporting this.  The PyTables project is largely in maintenance mode now, and this should be address rather sooner than later.  Currently, most of the developers are busy with other projects. Fortunately, we received a recent donation to the project via NumFOCUS (thanks ActiveState!) and I'd like to give priority to fix this by using these welcome resources.  Stay tuned!

**Comment by sw-yx** [comment]
this comment was most helpful for me https://github.com/PyTables/PyTables/issues/385#issuecomment-53220225 - make sure to try it

**Comment by iamhectorotero** [comment]
In my case, to make the ```pip install tables``` work in Linux in a Conda environment I had to previously install:
```
conda install h5py python-blosc python-lzo
```
Alternatively, (and without Conda) I could make it work with:
```
sudo apt install python-blosc python-lzo
pip install h5py
```
--------------------------------
If the ```python-lzo``` installation fails, try installing:
```
sudo apt install liblzo2-dev
```

**Comment by tomkooij** [comment]
@iamhectorotero : Do you really need to `pip install h5py` (or `conda install h5py`) ? I'm guessing this just installs the HDF5 lib as this is an h5py dependency.

Using conda try just installing the hdf5 package:

`conda install hd5`
or
`conda install -c conda-forge hdf5`

Without conda you need:

`apt-get install libhdf5-serial-dev`

**Comment by iamhectorotero** [comment]
Yeah that's possible. However, when I installed the `libhdf5-serial-dev` package through apt, `pytables` couldn't find the package's directory even if I pointed at it with `HDF5_DIR`. 

I guess installing `h5py` dealt with that as well. Maybe installing `hdf5` through Conda solves it too.

**Comment by tomkooij** [comment]
@iamhectorotero : Ok, thanks for clarifying that. (I'm not sure why that fails, as this is what we use on Travis CI). I'll try to update the docs about his.

**Comment by narun4sk** [comment]
I've stumbled upon the same problem on `Debian GNU/Linux 9.11 (stretch)` :disappointed: 

```
# dpkg -l libhdf5-dev libhdf5-serial-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                                                  Version                         Architecture                    Description
+++-=====================================================-===============================-===============================-================================================================================================================
ii  libhdf5-dev                                           1.10.0-patch1+docs-3+deb9u1     amd64                           Hierarchical Data Format 5 (HDF5) - development files - serial version
ii  libhdf5-serial-dev                                    1.10.0-patch1+docs-3+deb9u1     all                             transitional dummy package
```
```
# dpkg -L 'libhdf5-dev' | grep '\.h' | xargs dirname | uniq
/usr/include/hdf5/serial
```
```
# ldconfig -v | grep -i hdf5
ldconfig: Path `/lib/x86_64-linux-gnu' given more than once
ldconfig: Path `/usr/lib/x86_64-linux-gnu' given more than once
ldconfig: /lib/x86_64-linux-gnu/ld-2.24.so is the dynamic linker, ignoring

	libhdf5_serial_hl.so.100 -> libhdf5_serial_hl.so.100.0.0
	libhdf5_openmpi.so.100 -> libhdf5_openmpi.so.100.0.1
	libhdf5_mpich_hl.so.100 -> libhdf5_mpich_hl.so.100.0.0
	libhdf5_openmpi_hl.so.100 -> libhdf5_openmpi_hl.so.100.0.0
	libhdf5_hl_cpp.so.100 -> libhdf5_hl_cpp.so.100.0.0
	libhdf5_openmpihl_fortran.so.100 -> libhdf5_openmpihl_fortran.so.100.0.0
	libhdf5_mpichhl_fortran.so.100 -> libhdf5_mpichhl_fortran.so.100.0.0
	libhdf5_serialhl_fortran.so.100 -> libhdf5_serialhl_fortran.so.100.0.0
	libhdf5_openmpi_fortran.so.100 -> libhdf5_openmpi_fortran.so.100.0.1
	libhdf5_serial_fortran.so.100 -> libhdf5_serial_fortran.so.100.0.1
	libhdf5_serial.so.100 -> libhdf5_serial.so.100.0.1
	libhdf5_mpich.so.100 -> libhdf5_mpich.so.100.0.1
	libhdf5_mpich_fortran.so.100 -> libhdf5_mpich_fortran.so.100.0.1
	libhdf5_cpp.so.100 -> libhdf5_cpp.so.100.0.0
```
```
# pip show h5py
Name: h5py
Version: 2.10.0
Summary: Read and write HDF5 files from Python
Home-page: http://www.h5py.org
Author: Andrew Collette
Author-email: andrew.collette@gmail.com
License: BSD
Location: /home/jupyter/env/lib/python3.5/site-packages
Requires: numpy, six
Required-by: Keras, Keras-Applications
```
```
# pip install tables
Collecting tables
  Using cached https://files.pythonhosted.org/packages/2b/32/847ee3f521aae6a0be380d923a736162d698586f444df1ac24b98c65025c/tables-3.6.1.tar.gz
    ERROR: Command errored out with exit status 1:
     command: /home/jupyter/env/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-nfleihfw/tables/setup.py'"'"'; __file__='"'"'/tmp/pip-install-nfleihfw/tables/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-nfleihfw/tables/pip-egg-info
         cwd: /tmp/pip-install-nfleihfw/tables/
    Complete output (15 lines):
    /tmp/H5closev5alsgso.c:1:1: warning: return type defaults to ‘int’ [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/H5closev5alsgso.c: In function ‘main’:
    /tmp/H5closev5alsgso.c:2:5: warning: implicit declaration of function ‘H5close’ [-Wimplicit-function-declaration]
         H5close();
         ^~~~~~~
    /usr/bin/ld: cannot find -lhdf5
    collect2: error: ld returned 1 exit status
    * Using Python 3.5.3 (default, Sep 27 2018, 17:25:39)
    * USE_PKGCONFIG: False
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```
```
# export HDF5_DIR=/usr/include/hdf5/serial
# pip install tables
Collecting tables
  Using cached https://files.pythonhosted.org/packages/2b/32/847ee3f521aae6a0be380d923a736162d698586f444df1ac24b98c65025c/tables-3.6.1.tar.gz
    ERROR: Command errored out with exit status 1:
     command: /home/jupyter/env/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-i7ss9ow2/tables/setup.py'"'"'; __file__='"'"'/tmp/pip-install-i7ss9ow2/tables/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-i7ss9ow2/tables/pip-egg-info
         cwd: /tmp/pip-install-i7ss9ow2/tables/
    Complete output (15 lines):
    /tmp/H5closemrhyr3mp.c:1:1: warning: return type defaults to ‘int’ [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/H5closemrhyr3mp.c: In function ‘main’:
    /tmp/H5closemrhyr3mp.c:2:5: warning: implicit declaration of function ‘H5close’ [-Wimplicit-function-declaration]
         H5close();
         ^~~~~~~
    /usr/bin/ld: cannot find -lhdf5
    collect2: error: ld returned 1 exit status
    * Using Python 3.5.3 (default, Sep 27 2018, 17:25:39)
    * USE_PKGCONFIG: False
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```
```
# pip install --install-option="--hdf5=/usr/include/hdf5/serial" tables
/home/jupyter/env/lib/python3.5/site-packages/pip/_internal/commands/install.py:283: UserWarning: Disabling all use of wheels due to the use of --build-options / --global-options / --install-options.
  cmdoptions.check_install_build_global(options)
Collecting tables
  Using cached https://files.pythonhosted.org/packages/2b/32/847ee3f521aae6a0be380d923a736162d698586f444df1ac24b98c65025c/tables-3.6.1.tar.gz
    ERROR: Command errored out with exit status 1:
     command: /home/jupyter/env/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-g40jrxie/tables/setup.py'"'"'; __file__='"'"'/tmp/pip-install-g40jrxie/tables/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-g40jrxie/tables/pip-egg-info
         cwd: /tmp/pip-install-g40jrxie/tables/
    Complete output (15 lines):
    /tmp/H5closeol9u_rnj.c:1:1: warning: return type defaults to ‘int’ [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/H5closeol9u_rnj.c: In function ‘main’:
    /tmp/H5closeol9u_rnj.c:2:5: warning: implicit declaration of function ‘H5close’ [-Wimplicit-function-declaration]
         H5close();
         ^~~~~~~
    /usr/bin/ld: cannot find -lhdf5
    collect2: error: ld returned 1 exit status
    * Using Python 3.5.3 (default, Sep 27 2018, 17:25:39)
    * USE_PKGCONFIG: False
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```
```
# pip install git+https://github.com/PyTables/PyTables.git@master#egg=tables
Collecting tables
  Cloning https://github.com/PyTables/PyTables.git (to revision master) to /tmp/pip-install-p9op14h_/tables
  Running command git clone -q https://github.com/PyTables/PyTables.git /tmp/pip-install-p9op14h_/tables
    ERROR: Command errored out with exit status 1:
     command: /home/jupyter/env/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-p9op14h_/tables/setup.py'"'"'; __file__='"'"'/tmp/pip-install-p9op14h_/tables/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-p9op14h_/tables/pip-egg-info
         cwd: /tmp/pip-install-p9op14h_/tables/
    Complete output (15 lines):
    /tmp/H5closehmk_ewsg.c:1:1: warning: return type defaults to ‘int’ [-Wimplicit-int]
     main (int argc, char **argv) {
     ^~~~
    /tmp/H5closehmk_ewsg.c: In function ‘main’:
    /tmp/H5closehmk_ewsg.c:2:5: warning: implicit declaration of function ‘H5close’ [-Wimplicit-function-declaration]
         H5close();
         ^~~~~~~
    /usr/bin/ld: cannot find -lhdf5
    collect2: error: ld returned 1 exit status
    * Using Python 3.5.3 (default, Sep 27 2018, 17:25:39)
    * USE_PKGCONFIG: False
    .. ERROR:: Could not find a local HDF5 installation.
       You may need to explicitly state where your local HDF5 headers and
       library can be found by setting the ``HDF5_DIR`` environment
       variable or by using the ``--hdf5`` command-line option.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

Is there anything else I may try?

**Comment by iamhectorotero** [comment]
Indeed @tomkooij, using:
```
conda install hdf5
```
as you pointed out instead of my suggestion (installing `h5py` `python-blosc` and `python-lzo`) solves the issue as well.

PS. Sorry for the ~8m reply.
